# **<u>FOOD DELIVERY COST AND PROFITABILITY ANALYSIS</u>**

<span style="color: orange; font-style: italic;">Maureen Ndunge Kitang'a</span>

## **PROJECT PROPOSAL**
### *Executive Summary*

Our analysis of Uber Eats data is geared towards extracting valuable insights into customer preferences within the food industry. We're looking at how much restaurants charge, their ratings, the types of food they offer, and where they're located. Our aim? Provide straightforward tips to help these places get noticed more on the platform and make more money.

Analyzing this data helps us spot trends and patterns in the food scene. Our main goal is to provide practical recommendations that can reshape restaurant strategies and enhance their visibility on the Uber Eats platform.

### *Problem Statement*
Our main mission in this project is to provide practical guidance to clients planning to start a new restaurant chain or enhance the performance of their existing establishments. We're dealing with a widespread issue in the restaurant industry, where profit margins typically fall within the range of 10-20%. Our analysis is centered around exploring the current landscape of Uber Eats, aiming to uncover inventive strategies that can make restaurant businesses more attractive to both new and existing customers, ultimately driving higher profits.

# **WHY DOES THIS MATTER**
The significance of this problem stems from the remarkable growth of online food ordering platforms, exemplified by Uber Eats' substantial transaction surge. In 2022, Uber Eats recorded a staggering USD 11 billion in revenue, marking a notable 31% increase from the previous year's revenue of $8.3 billion. Concurrently, there has been a steady 2% growth in user numbers, with a significant 10% increase in merchant participation in the US. These trends underscore the platform's rising popularity among both customers and merchants.

Central to this challenge is understanding customer preferences across various dimensions, including preferred cuisines and menu diversity, within different regions. Thus, the pivotal question emerges: How can restaurants effectively analyze customer preferences to craft strategies that capitalize on the burgeoning potential of online food ordering platforms?

# **DATA UNDERSTANDING**
The primary dataset utilized in this analysis has information on various restaurants spread across the United States. Data sources were obtained through web scraping collected using Python libraries and the Uber Eats website.
There are 2 datasets - Restaurants dataset, and the Menus dataset. For more in information on the [data]("https://www.kaggle.com/datasets/ahmedshahriarsakib/uber-eats-usa-restaurants-menus")

In [12]:
#importing the necessary libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


### *Dataset 1:Restaurants Dataset*

In [13]:
#Loading data into a Pandas DataFrame
restaurants = pd.read_csv("./data/restaurants.csv")
#Previewing the first few rows
restaurants.head()

id  position                                            name  score  \
0   1        19               PJ Fresh (224 Daniel Payne Drive)    NaN   
1   2         9                  J' ti`'z Smoothie-N-Coffee Bar    NaN   
2   3         6  Philly Fresh Cheesesteaks (541-B Graymont Ave)    NaN   
3   4        17         Papa Murphy's (1580 Montgomery Highway)    NaN   
4   5       162                Nelson Brothers Cafe (17th St N)    4.7   

   ratings                                          category price_range  \
0      NaN                     Burgers, American, Sandwiches           $   
1      NaN  Coffee and Tea, Breakfast and Brunch, Bubble Tea         NaN   
2      NaN        American, Cheesesteak, Sandwiches, Alcohol           $   
3      NaN                                             Pizza           $   
4     22.0         Breakfast and Brunch, Burgers, Sandwiches         NaN   

                                        full_address zip_code        lat  \
0      224 Daniel Payne Drive, Birmingham, AL, 35207    35207  33.562365   
1  1521 Pinson Valley Parkway, Birmingham, AL, 35217    35217  33.583640   
2          541-B Graymont Ave, Birmingham, AL, 35204    35204  33.509800   
3         1580 Montgomery Highway, Hoover, AL, 35226    35226  33.404439   
4               314 17th St N, Birmingham, AL, 35203    35203  33.514730   

         lng  
0 -86.830703  
1 -86.773330  
2 -86.854640  
3 -86.806614  
4 -86.811700

In [14]:
#Shape of the dataframe
print("The number of rows: {}".format(restaurants.shape[0]))

print("The number of columns:{}".format(restaurants.shape[1]))

The number of rows: 63469
The number of columns:11


In [15]:
#General information about the dataset
restaurants.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 63469 entries, 0 to 63468
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   id            63469 non-null  int64  
 1   position      63469 non-null  int64  
 2   name          63469 non-null  object 
 3   score         35302 non-null  float64
 4   ratings       35302 non-null  float64
 5   category      63384 non-null  object 
 6   price_range   52852 non-null  object 
 7   full_address  63016 non-null  object 
 8   zip_code      62952 non-null  object 
 9   lat           63469 non-null  float64
 10  lng           63469 non-null  float64
dtypes: float64(4), int64(2), object(5)
memory usage: 5.3+ MB


The restaurants dataset 63469 rows and 11 columns. The columns with missing data are `score`, `ratings`, `category`, `price_range`, `full_address`,and `zip_code`

### *Dataset 2:Menus Dataset*

In [16]:
#Loading the data into the Pandas DataFrame
menus = pd.read_csv("./data/restaurant-menus.csv")
#Previewing the last few rows
menus.tail()

restaurant_id          category                             name  \
5117212          63469  Other Essentials             Composition Notebook   
5117213          63469  Other Essentials   Fancy Fest Savory Salmon - 3oz   
5117214          63469  Other Essentials            Bicycle Playing Cards   
5117215          63469  Other Essentials           Tidy Cat Liter - 10lbs   
5117216          63469  Other Essentials  7-Select Heavy Duty Foam Cooler   

        description     price  
5117212         NaN  4.38 USD  
5117213         NaN  1.19 USD  
5117214         NaN  3.83 USD  
5117215         NaN  4.38 USD  
5117216         NaN   6.8 USD

In [17]:
#Shape of the dataframe
print("The number of rows: {}".format(menus.shape[0]))

print("The number of columns:{}".format(menus.shape[1]))

The number of rows: 5117217
The number of columns:5


In [18]:
#General information about the dataset
menus.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5117217 entries, 0 to 5117216
Data columns (total 5 columns):
 #   Column         Dtype 
---  ------         ----- 
 0   restaurant_id  int64 
 1   category       object
 2   name           object
 3   description    object
 4   price          object
dtypes: int64(1), object(4)
memory usage: 195.2+ MB


# **DATA PREPARATION**

During the Data Preparation phase, we will be performing a series of essential tasks to prepare our raw data for analysis. This phase includes the following key activities:

*Merging Datasets:*
We will combine multiple datasets, if available, to create a comprehensive dataset that encompasses all relevant information for our analysis. This may involve joining datasets based on common keys or merging them using appropriate techniques.

*Deriving New Attributes:*
To enhance the richness of our dataset and capture additional insights, we will create new attributes or features through feature engineering. This process involves transforming existing variables, generating new variables, or extracting valuable information from the data.

*Data Cleaning:*
Data cleaning is a crucial step that involves identifying and addressing various data quality issues, such as missing values, outliers, duplicates, and inconsistencies. We will employ techniques such as imputation, deletion, outlier detection, and data validation to ensure the integrity and quality of our dataset.

*Exploratory Data Analysis (EDA):*
EDA plays a vital role in understanding the underlying patterns, trends, and relationships within our data. We will perform exploratory data analysis to visualize distributions, examine correlations, detect patterns, and gain insights into the characteristics of our dataset. This will guide our subsequent analysis and decision-making processes.

### **DATA CLEANING**
>> We will begin our Data Preparation phase by prioritizing data cleaning for individual datasets before merging. This approach allows us to address data inconsistencies, missing values, duplicates, outliers, and other quality issues specific to each dataset. By cleaning the datasets individually, we can ensure data integrity and consistency before merging. Additionally, considering differences in data sizes and structures among datasets, cleaning them separately facilitates more focused and efficient data cleaning efforts. Once each dataset is cleaned and standardized, we will proceed with merging them to create a comprehensive dataset for further analysis.

In [19]:
def check_missing_values(df):
    """
    Check for missing values in each column of a DataFrame.

    Parameters:
    df (DataFrame): The DataFrame to check for missing values.

    Returns:
    missing_values (DataFrame): A DataFrame showing the count and percentage of missing values in each column.
    """
    missing_count = df.isnull().sum()  # Count missing values in each column
    missing_percentage = (missing_count / len(df)) * 100  # Calculate percentage of missing values

    # Create DataFrame to display missing values count and percentage
    missing_values = pd.DataFrame({
        'Missing Values': missing_count,
        'Percentage': missing_percentage.round(2)
    })
    
    return missing_values

In [20]:
check_missing_values(restaurants)

Missing Values  Percentage
id                         0        0.00
position                   0        0.00
name                       0        0.00
score                  28167       44.38
ratings                28167       44.38
category                  85        0.13
price_range            10617       16.73
full_address             453        0.71
zip_code                 517        0.81
lat                        0        0.00
lng                        0        0.00

The columns '`id`, `position`, `name`, `lat`, and `lng` have no missing values (0% missing). These columns are complete and do not require imputation or further handling for missing data.

The columns `score` and `ratings` have a significant proportion of missing values, with approximately `44.38%`missing in each column. This suggests that a large portion of the data in these columns is missing.Given the significance of these features and their importance for the analysis, median imputation will be done to handle the missing values.

The columns `category`, `price_range`, `full_address`, and `zip_code` have relatively fewer missing values, ranging from 0.13% to 16.73%, it's worth considering whether these missing values are significant enough to warrant deletion.

In [21]:
check_missing_values(menus)

Missing Values  Percentage
restaurant_id               0        0.00
category                    0        0.00
name                        4        0.00
description           1452145       28.38
price                       0        0.00

The `description` column stands out as having a substantial number of missing values. To address this issue, rows with missing values in the 'description' column will be deleted to ensure data integrity.
The other columns (`restaurant_id`, `category`, '`name`, and `price`) have either no missing values or a negligible number of missing values, suggesting that they are relatively complete and may not require extensive handling for missing data.